app/controllers/application_controller.rb

```diff
@@ -26,6 +27,25 @@ def authorize(record, query=nil, user=nil)
     end
   end
 
+  def param_date(sym)
+    begin
+      Date.strptime(params[sym].to_s)
+    rescue
+      nil
+    end
+  end
+
+  def return_to
+    if params[:return_to]
+      return_to = params.require(:return_to)
+      proper_sig = self.make_return_to(return_to.require(:name),
+                                       return_to.require(:path))[:sig]
+      if return_to.require(:sig) == proper_sig
+        return_to.permit([:path, :name])
+      end
+    end
+  end
```

Comment:
I'd like to leave it + test it. I do remember one of our testers not having any clue how to get back. This is a pretty lightweight way of offering that. That said, we may find that longer lists will require that this link be placed at the bottom, but placement is good as the first pass.